src/pages/ProfilesExplorerView/components/SceneExploreDiffFlameGraph/components/ScenePresetsPicker/ScenePresetsPicker.tsx

```diff
@@ -0,0 +1,273 @@
+import { css } from '@emotion/css';
+import { GrafanaTheme2, SelectableValue } from '@grafana/data';
+import {
+  SceneComponentProps,
+  sceneGraph,
+  SceneObjectBase,
+  SceneObjectState,
+  VariableDependencyConfig,
+} from '@grafana/scenes';
+import { Button, Modal, Select, useStyles2 } from '@grafana/ui';
+import { reportInteraction } from '@shared/domain/reportInteraction';
+import React from 'react';
+
+import { FEEDBACK_FORM_URL } from '../../../GiveFeedbackButton';
+import { CompareTarget } from '../../../SceneExploreServiceLabels/components/SceneGroupByLabels/components/SceneLabelValuesGrid/domain/types';
+import { SceneComparePanel } from '../SceneComparePanel/SceneComparePanel';
+
+interface ScenePresetsPickerState extends SceneObjectState {
+  name: string;
+  label: string;
+  isModalOpen: boolean;
+  isSelectOpen: boolean;
+  value: string | null;
+}
+
+export type Preset = {
+  from: string;
+  to: string;
+  diffFrom: string;
+  diffTo: string;
+  label: string;
+};
+
+export class ScenePresetsPicker extends SceneObjectBase<ScenePresetsPickerState> {
+  protected _variableDependency = new VariableDependencyConfig(this, {
+    variableNames: ['dataSource', 'serviceName'],
+    onReferencedVariableValueChanged: () => {
+      this.reset();
+    },
+  });
+
+  static PRESETS = [
+    {
+      label: 'Built-in presets',
+      value: 'built-in',
+      options: [
+        {
+          value: '1h ago vs now (30m)',
+          label: 'Last hour (30m-window)',
+          baseline: {
+            from: 'now-1h',
+            to: 'now',
+            diffFrom: 'now-1h',
+            diffTo: 'now-30m',
+            label: 'last hour',
+          },
+          comparison: {
+            from: 'now-1h',
+            to: 'now',
+            diffFrom: 'now-30m',
+            diffTo: 'now',
+            label: 'last hour',
+          },
+        },
+        {
+          value: '1h ago vs now (1h)',
+          label: 'Last hour (1h-window)',
+          baseline: {
+            from: 'now-1h',
+            to: 'now',
+            diffFrom: 'now-1h',
+            diffTo: 'now',
+            label: 'last hour',
+          },
+          comparison: {
+            from: 'now-1h',
+            to: 'now',
+            diffFrom: 'now-1h',
+            diffTo: 'now',
+            label: 'last hour',
+          },
+        },
+        {
+          value: '6h ago vs now',
```

Comment:
**question**: Is the rane intentional?

It selects the following (at 11:23):

<img width="1689" alt="Screenshot 2024-10-29 at 11 23 47" src="https://github.com/user-attachments/assets/ccec7d43-b70b-4803-b369-140af08a2439">

So it compares 6h-5.5h with 0.5h-0h ago

So it it was 12:00 it would compare: 6:00-6:30 with 11:30-12:00. I was kind of expecting it would compare 5:30-6:00 with 11:30-12:00 (so the "to" is shifted by 6 hours, not 5.5h. 

Same with the 24h preset below. I'm not sure what makes more sense for users, so maybe it's good the way it is. Just wanted to double check.

Reply:
Good question: maybe it would make more sense to center the (e.g.) 6h-mark.
I.e. selecting -6h15 to -5h45 so that the user sees 30m around the 6h mark. Having said this, I find it a bit arbitrary... I doubt that our users will select a preset to be precise about time, but more like a good approximation to start adjusting the desired range.

Reply:
I made the changes